### PR TITLE
Happy API

### PR DESCRIPTION
**Pull request by LillanEriksson**
https://nervous-mayer-699099.netlify.app/

https://my-happy-api.herokuapp.com/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,29 +1,98 @@
...
+		});
+	}
+});
+
+//Like a post
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+	try {
+		await Thought.updateOne(
+			{ _id: req.params.thoughtId },
```
Comment (JHarde) [inline server.js:83]: Good use of req.params! 👍

File: server.js [hunk 2]
```diff
@@ -1,29 +1,98 @@
...
+});
+
+//Post a new thought
+
+app.post('/thoughts', async (req, res) => {
+	const { message } = req.body;
+	const newThought = await new Thought({ message });
+	try {
+		const savedThought = await newThought.save();
+		res.status(201).json(savedThought);
```
Comment (JHarde) [inline server.js:68]: Nice job with the status messages, makes the code easy to read and understand! :)

File: server.js [hunk 3]
```diff
@@ -1,29 +1,98 @@
...
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+	res.send('Welcome to Happy happy place');
+});
+
+app.get('/thoughts', async (req, res) => {
+	try {
+		const thoughts = await Thought.find()
+			.sort({ createdAt: 'desc' })
```
Comment (JHarde) [inline server.js:47]: Good sorting of the messages and clear error message! 🎖️

File: server.js [hunk 4]
```diff
@@ -1,29 +1,98 @@
...
+		type: Date,
+		default: Date.now,
+	},
+});
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+	res.send('Welcome to Happy happy place');
```
Comment (JHarde) [inline server.js:41]: Really like the message!

File: server.js [hunk 5]
```diff
@@ -1,29 +1,98 @@
...
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+//Mongoose model
+const Thought = new mongoose.model('Thought', {
```
Comment (JHarde) [inline server.js:22]: Good looking model! 🏅

**Review by JHarde — APPROVED** [review]
Great job Lillemor! Didn't really find much you need to improve, your code is very clean and easy to read and understand, keep up the good work! 💯

File: server.js [hunk 6]
```diff
@@ -1,29 +1,98 @@
...
+			.limit(20)
+			.exec();
+
+		res.json(thoughts);
+	} catch (err) {
+		res.status(400).json({
+			success: false,
+			message: 'Could not get thoughts',
+			errors: err.errors,
+		});
```
Comment (KarolCislowski) [inline server.js:57]: nice constructed error json 👍🏻

**Review by KarolCislowski — APPROVED** [review]
This is a very nice project 😸 

Please revisit the `README` file and give it some love 🥰

**Pull request closed without merging**